src/modules/TokenGateway.sol

```diff
@@ -19,7 +19,7 @@ error MinterRoleMissing();
 error AuthFailed();
 error NotDispatcher();
 
-contract TokenGateway is IIsmpModule {
+abstract contract TokenGateway is IIsmpModule {
```

Comment:
why abstract?

Reply:
Some functions are not implemented. I will update that in a bit

Reply:
it's a complex topic, recommend you research more about it:

https://fravoll.github.io/solidity-patterns/tight_variable_packing.html